backpack/extensions/secondorder/diag_ggn/__init__.py

```diff
@@ -136,3 +136,83 @@ def __init__(self, mc_samples=1):
+
+
+class BatchDiagGGN(BackpropExtension):
+    """
+    Individual Diagonal of the Generalized Gauss-Newton/Fisher.
+    Uses the exact Hessian of the loss w.r.t. the model output.
+
+    Stores the output in :code:`diag_ggn_exact_batch`,
+    has the same dimensions as the gradient.
+
+    For a faster but less precise alternative,
+    see :py:meth:`backpack.extensions.DiagGGNMC`.
+
+    """
+
+    VALID_LOSS_HESSIAN_STRATEGIES = [
+        LossHessianStrategy.EXACT,
+        LossHessianStrategy.SAMPLING,
+    ]
+
+    def __init__(self, loss_hessian_strategy=LossHessianStrategy.EXACT, savefield=None):
+        if savefield is None:
+            savefield = "diag_ggn_batch"
+        if loss_hessian_strategy not in self.VALID_LOSS_HESSIAN_STRATEGIES:
+            raise ValueError(
+                "Unknown hessian strategy: {}".format(loss_hessian_strategy)
+                + "Valid strategies: [{}]".format(self.VALID_LOSS_HESSIAN_STRATEGIES)
+            )
+
+        self.loss_hessian_strategy = loss_hessian_strategy
+        super().__init__(
+            savefield=savefield,
+            module_exts={
+                MSELoss: losses.DiagGGNMSELoss(),
+                CrossEntropyLoss: losses.DiagGGNCrossEntropyLoss(),
+                Linear: linear.BatchDiagGGNLinear(),
+                MaxPool1d: pooling.DiagGGNMaxPool1d(),
+                MaxPool2d: pooling.DiagGGNMaxPool2d(),
+                AvgPool1d: pooling.DiagGGNAvgPool1d(),
+                MaxPool3d: pooling.DiagGGNMaxPool3d(),
+                AvgPool2d: pooling.DiagGGNAvgPool2d(),
+                AvgPool3d: pooling.DiagGGNAvgPool3d(),
+                ZeroPad2d: padding.DiagGGNZeroPad2d(),
+                Conv1d: conv1d.BatchDiagGGNConv1d(),
+                Conv2d: conv2d.BatchDiagGGNConv2d(),
+                Conv3d: conv3d.BatchDiagGGNConv3d(),
+                ConvTranspose1d: convtranspose1d.BatchDiagGGNConvTranspose1d(),
+                ConvTranspose2d: convtranspose2d.BatchDiagGGNConvTranspose2d(),
+                ConvTranspose3d: convtranspose3d.BatchDiagGGNConvTranspose3d(),
+                Dropout: dropout.DiagGGNDropout(),
+                Flatten: flatten.DiagGGNFlatten(),
+                ReLU: activations.DiagGGNReLU(),
+                Sigmoid: activations.DiagGGNSigmoid(),
+                Tanh: activations.DiagGGNTanh(),
+                LeakyReLU: activations.DiagGGNLeakyReLU(),
+                LogSigmoid: activations.DiagGGNLogSigmoid(),
+                ELU: activations.DiagGGNELU(),
+                SELU: activations.DiagGGNSELU(),
+            },
+        )
+
+
+class BatchDiagGGNExact(BatchDiagGGN):
+    """
+    Diagonal of the Generalized Gauss-Newton/Fisher.
+    Uses the exact Hessian of the loss w.r.t. the model output.
+
+    Stores the output in :code:`diag_ggn_exact`,
+    has the same dimensions as the gradient.
+
+    For a faster but less precise alternative,
+    see :py:meth:`backpack.extensions.DiagGGNMC`.
+
+    """
```

Comment:
I added remarks above how to refactor the docstring of `BatchDiagGGN`. After that, move the docstring from `BatchDiagGGN` to this place such that `BatchDiagGGN`'s docstring is empty.